### PR TITLE
Add time slice fairness test

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- add test to cover time-based queue fairness when `max_high_before_low` is zero
- clean up duplicate imports in test module

## Testing
- `make fmt`
- `make lint`
- `make test`
- `cargo test fairness_yields_low_with_time_slice -- --nocapture`

------
https://chatgpt.com/codex/tasks/task_e_68684cd037548322a494feb0d1e8476d

## Summary by Sourcery

Add a test to verify time-based queue fairness with a nonzero time slice when high-priority quotas are zero and remove redundant imports from the test module

Tests:
- Introduce `fairness_yields_low_with_time_slice` to ensure low-priority items are yielded based on time slices when `max_high_before_low` is zero

Chores:
- Clean up duplicate imports in the connection actor test module

### CODE REVIEW

File: tests/connection_actor.rs [hunk 1]
```diff
@@ -68,6 +71,48 @@ async fn fairness_yields_low_after_burst(
...
+    drop(handle);
+
+    let out = rx.await.unwrap();
+    assert!(out.contains(&42), "Low-priority item was not yielded");
+    let pos = out.iter().position(|x| *x == 42).unwrap();
+    assert!(
+        pos > 0 && pos < out.len() - 1,
+        "Low-priority item should be yielded in the middle"
+    );
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/connection_actor.rs:147]: @coderabbitai would tokio::time::pause) help here?

[the remark this answers is not in the document]
Comment (leynos) [reply tests/connection_actor.rs:147]: @coderabbitai Please write a prompt for a coding assistant proposing the `pause` solution
